src/ApplicationInsights.ServiceFabric.Native/Net45/IWrappingClient.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Microsoft.ApplicationInsights.ServiceFabric.Remoting.Activities
+{
+    using Microsoft.ServiceFabric.Services.Remoting.Client;
+
+    /// <summary>
+    /// Interface implemented by remoting clients to provide access to the inner client.
+    /// </summary>
+    internal interface IWrappingClient
```

Comment:
The requirement is driven by CorrelatingServiceRemotingClientFactory, since it has this code:

        public Task<OperationRetryControl> ReportOperationExceptionAsync(IServiceRemotingClient client, ExceptionInformation exceptionInformation, OperationRetrySettings retrySettings, CancellationToken cancellationToken)
        {
            IServiceRemotingClient effectiveClient = (client as IWrappingClient)?.InnerClient ?? client;
            return this.innerClientFactory.ReportOperationExceptionAsync(effectiveClient, exceptionInformation, retrySettings, cancellationToken);
        }

You cannot change the signature of ReportOperationExceptionAsync because it's defined by ICommunicationClientFactory<IServiceRemotingClient>, which IServiceRemotingClientFactory derives from. What I meant was, ReportOperationExceptionAsync can instead downcast it to the concrete client type instead of the interface type. I am fine either way.

Of course, if the factory doesn't need access to InnerClient, then the correlating client can implement it using a field like you suggested.